app/jobs/batch_create_job.rb

```diff
@@ -1,63 +1,33 @@
 class BatchCreateJob < ActiveJob::Base
-  include Hydra::PermissionsQuery
-
   queue_as :batch_create
 
-  attr_accessor :saved, :success
+  before_enqueue do |job|
+    log = job.arguments.last
+    log.pending_job(self)
+  end
 
   # This copies metadata from the passed in attribute to all of the works that
   # are members of the given upload set
-  def perform(user, titles, resource_types, uploaded_files, attributes)
-    @saved = []
-    @success = true
+  def perform(user, titles, resource_types, uploaded_files, attributes, log)
+    log.performing!
 
     titles ||= {}
     resource_types ||= {}
 
-    create(user, titles, resource_types, uploaded_files, attributes)
-    send_user_message(user)
```

Comment:
It's moved to the BatchCreateOperation, and it occurs when all of the parts are done.